scijava-ops-engine/src/main/java/org/scijava/ops/engine/matcher/impl/DefaultOpMatcher.java

```diff
@@ -71,17 +71,24 @@ public OpCandidate match(MatchingConditions conditions, OpEnvironment env) {
 		}
 
 		// in the case of no matches, throw an agglomerated exception
-		throw agglomeratedException(exceptions);
+		throw agglomeratedException(conditions.request(), exceptions, env);
 	}
 
-	private OpMatchingException agglomeratedException(
-		List<OpMatchingException> list)
-	{
-		OpMatchingException agglomerated = new OpMatchingException(
-			"No MatchingRoutine was able to produce a match!");
-		for (int i = 0; i < list.size(); i++) {
-			agglomerated.addSuppressed(list.get(i));
+	private OpMatchingException agglomeratedException( //
+		final OpRequest request, //
+		final List<OpMatchingException> list, //
+		final OpEnvironment env //
+	) {
+		// Develop help message
+		var msg = "No match found!";
+		try {
+			// TODO: Remove try/catch.
+			msg += " Perhaps you meant: \n" + env.help(request);
```

Comment:
conceptually this seems useful. We'll have to see if it is in practice but I don't see any reason not to do it now.